shared/question/question.ts

```diff
@@ -48,7 +48,8 @@ class Question {
     }
 
     isAnswerValid() {
-        return !(this.required && this.answer === undefined);
+        return !(this.required && (this.answer === undefined || (this.answer.length !== undefined && !this.answer.length))
```

Comment:
```suggestion
        return !(this.required && (this.answer === undefined || this.answer.length === 0))
```
But I think this check should be only on question types which their answer's type is an array (`CheckboxQuestion`)

Reply:
note that this is outdated

Reply:
This is not good since if answer is undefined, it still call answer.length and then errors "no property length of undefined" 
I don't know why it checks them both, but it is :( 
This is why I did that